libraries/tracker-core/src/emitter/index.ts

```diff
@@ -0,0 +1,410 @@
+   * Only non-success status codes are considered (greater or equal to 300).
+   * The retry codes are only considered for GET and POST requests.
+   * By default, the tracker retries on all non-success status codes except for 400, 401, 403, 410, and 422.
+   */
+  retryStatusCodes?: number[];
+  /**
+   * List of HTTP response status codes for which events sent to Collector should not be retried in future request.
+   * Only non-success status codes are considered (greater or equal to 300).
+   * The don't retry codes are only considered for GET and POST requests.
+   * By default, the tracker retries on all non-success status codes except for 400, 401, 403, 410, and 422.
+   */
+  dontRetryStatusCodes?: number[];
+  /**
+   * Id service full URL. This URL will be added to the queue and will be called using a GET method.
+   * This option is there to allow the service URL to be called in order to set any required identifiers e.g. extra cookies.
+   *
+   * The request respects the `anonymousTracking` option, including the SP-Anonymous header if needed, and any additional custom headers from the customHeaders option.
+   */
+  idService?: string;
+  /**
+   * Indicates that the request should be allowed to outlive the webpage that initiated it.
+   * Enables collector requests to complete even if the page is closed or navigated away from.
+   * Defaults to false.
+   */
+  keepalive?: boolean;
+  /**
+   * Enables overriding the default fetch function with a custom implementation.
+   * @param input - Instance of Request
+   * @param options - Additional options for the request
+   * @returns A Promise that resolves to the Response.
+   */
+  customFetch?: (input: Request, options?: RequestInit) => Promise<Response>;
+  /**
+   * A callback function to be executed whenever a request is successfully sent to the collector.
+   * In practice this means any request which returns a 2xx status code will trigger this callback.
+   *
+   * @param data - The event batch that was successfully sent
+   */
+  onRequestSuccess?: (data: EventBatch) => void;
+  /**
+   * A callback function to be executed whenever a request fails to be sent to the collector.
+   * This is the inverse of the onRequestSuccess callback, so any non 2xx status code will trigger this callback.
+   *
+   * @param data - The data associated with the event(s) that failed to send
+   */
+  onRequestFailure?: (data: RequestFailure) => void;
+  /**
+   * Enables providing a custom EventStore implementation to store events before sending them to the collector.
+   */
+  eventStore?: EventStore;
+}
+
+export interface EmitterConfiguration extends EmitterConfigurationBase {
+  /* The collector URL to which events will be sent */
+  endpoint: string;
+  /* http or https. Defaults to https */
+  protocol?: 'http' | 'https';
+  /* Collector port number */
+  port?: number;
+  /* If the request should undergo server anonymization. */
+  serverAnonymization?: boolean;
```

Comment:
Yeah, I think you're correct that this is a good chance to solve this old problem since we are making a breaking change to the event storage. I have added a `EventStorePayload` wrapper around the `Payload` that adds this server anonymization information, hopefully this will solve it!